### PR TITLE
[5.1] Menus and categories view with equal headline

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue # .

### Summary of Changes
Make headline in categories like headline in menus. There is plenty of place for text and is better a11y and usability.

### Testing Instructions
Compare menus and categories view in backend

### Actual result BEFORE applying this Pull Request

grafik

grafik

### Expected result AFTER applying this Pull Request
grafik

### CODE REVIEW

**Comment by brianteeman** [comment]
If this change is going to be made thn it needs to be made everywhere - I know of at least banners-clients that would also need to be updated - there may be others.

Also its not a bug fix so it should be in 4.3 i think

**Comment by chmst** [comment]
agree. 
Do you see any problems there? It is better UX and better a11y.

**Comment by brianteeman** [comment]
Not sure myself that it's better anything, but it must be at least consistent

**Comment by crystalenka** [comment]
I think this is definitely more clear, at least in desktop. What happens in mobile? 

consistency is of course important, but let's figure out the best pattern and then we can see where else it needs to be implemented. :)

**Comment by chmst** [comment]
Ithas to be made equal for all categories. Now w have inconsitency. WhyI prefere this versio: Now we  make  text  invisible when there is plenty of place, no need for ugly titles. The dropdown for hiding columns says "archived items"  - is it obvious that the folder icon stands tor archived items? 
You know that of course - but a new user?

**Comment by brianteeman** [comment]
will you be updating banner-clients?

**Comment by chmst** [comment]
There are all categories views - banner, newsfeed, contacts  (also weblinks). Maybe user notes .. I must investigate.

If the pattern is accepted then I can do it (but after my holidays). I can set this on draft, so nothing happens befor all is consistent.

**Comment by brianteeman** [comment]
> There are all categories views - banner, newsfeed, contacts (also weblinks). Maybe user notes .. I must investigate.

I checked them

**Comment by coolcat-creations** [comment]
I have tested this item :white_check_mark: successfully on c82adbf251d90757b4f0830f948cc5e7ea668a81<br /><br />Tested this successfully, thank you!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by crystalenka** [comment]
Tested successfully, but not submitting it on the tracker because I am reconsidering my earlier comment after seeing it in different components. :/ 

The labels are indeed more clear than only icons, but I am currently on my laptop which has a smaller screen and in some components, this change means that there is significantly less white space and a lot of visual weight is given to these counters when they're not necessarily the most critical part of these views:

Banners: Clients screen

Smart search: content maps

Articles: categories

As a result, even though the labels are more clear, the interface is harder to understand because the eye doesn't know where to go.

I agree both that the original pattern isn't very clear and that it should be consistent across all views like this, but I think we need to take a step back and find a different approach so it can be both clear and meaningful.

I was wrong before to make an opinion without seeing how it affected other components; apologies that you spent additional time on it just for me to come in and suggest a different direction. :(

**Comment by brianteeman** [comment]
@crystalenka thats why they were not given heading texts originally. It would be good if people remembered that previous work was not done on a whim without any thought.

**Comment by crystalenka** [comment]
> @crystalenka thats why they were not given heading texts originally. It would be good if people remembered that previous work was not done on a whim without any thought.

No one implied that. Previous work is important; we also need to be open to improving and getting better as we learn and grow. I think there is a solution that solves both problems but need to think through it more.

**Comment by chmst** [comment]
The smart search content map was not changed in this PR. I would remove the "indexed content" from the language string. 
I can do that if you want. 

In any case, these informations schould be consistent. We have 

- the "verbose" version in com_menus
- the "very verbose" version in smart search
- the "icons only" in categories and banners

So we were not consequent in the past. 

But if the JXT wants to make a concept which meets all needs, I will be happy.

File: administrator/components/com_banners/tmpl/clients/default.php [hunk 1]
```diff
@@ -72,22 +72,30 @@
...
+                                <?php endif; ?>
+                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_unpublished')) : ?>
+                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
+                                        <span class="icon-times" aria-hidden="true""></span>
+                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_BANNERS_COUNT_UNPUBLISHED_ITEMS'); ?></span>
+                                    </th>
+                                <?php endif; ?>
+                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
+                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
+                                        <span class="icon-folder icon-fw" aria-hidden="true""></span>
```
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php:89]:
```suggestion
                                        <span class="icon-times" aria-hidden="true""></span>
                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_BANNERS_COUNT_UNPUBLISHED_ITEMS'); ?></span>
                                    </th>
                                <?php endif; ?>
                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
                                        <span class="icon-folder icon-fw" aria-hidden="true""></span>
```
```suggestion
                                        <span class="icon-times" aria-hidden="true""></span>
                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_BANNERS_COUNT_UNPUBLISHED_ITEMS'); ?></span>
                                    </th>
                                <?php endif; ?>
                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
                                        <span class="icon-folder icon-fw" aria-hidden="true"></span>
```
```suggestion
                                        <span class="icon-times" aria-hidden="true""></span>
                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_BANNERS_COUNT_UNPUBLISHED_ITEMS'); ?></span>
                                    </th>
                                <?php endif; ?>
                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
                                        <span class="icon-folder icon-fw" aria-hidden="true"></span>
```

File: administrator/components/com_categories/tmpl/categories/default.php [hunk 2]
```diff
@@ -84,35 +84,35 @@
...
+                                        <span class="icon-check" aria-hidden="true"></span>
+                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_CATEGORIES_HEADING_PUBLISHED'); ?></span>
                                     </th>
                                 <?php endif; ?>
                                 <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_unpublished')) : ?>
-                                    <th scope="col" class="w-3 text-center d-none d-md-table-cell">
-                                        <span class="icon-times" aria-hidden="true" title="<?php echo Text::_('COM_CATEGORY_COUNT_UNPUBLISHED_ITEMS'); ?>"></span>
-                                        <span class="visually-hidden"><?php echo Text::_('COM_CATEGORY_COUNT_UNPUBLISHED_ITEMS'); ?></span>
+                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
+                                        <span class="icon-times" aria-hidden="true""></span>
```
Comment (Quy) [inline administrator/components/com_categories/tmpl/categories/default.php]:
```suggestion
                                        <span class="icon-times" aria-hidden="true"></span>
```

File: administrator/components/com_categories/tmpl/categories/default.php [hunk 3]
```diff
@@ -84,35 +84,35 @@
...
+                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_CATEGORIES_HEADING_UNPUBLISHED'); ?></span>
                                     </th>
                                 <?php endif; ?>
                                 <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
-                                    <th scope="col" class="w-3 text-center d-none d-md-table-cell">
-                                        <span class="icon-folder icon-fw" aria-hidden="true" title="<?php echo Text::_('COM_CATEGORY_COUNT_ARCHIVED_ITEMS'); ?>"></span>
-                                        <span class="visually-hidden"><?php echo Text::_('COM_CATEGORY_COUNT_ARCHIVED_ITEMS'); ?></span>
+                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
+                                        <span class="icon-folder icon-fw" aria-hidden="true""></span>
```
Comment (Quy) [inline administrator/components/com_categories/tmpl/categories/default.php]:
```suggestion
                                        <span class="icon-folder icon-fw" aria-hidden="true"></span>
```

File: administrator/components/com_banners/tmpl/clients/default.php (hunk 1 above)
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php]:
```suggestion
                                        <span class="icon-times" aria-hidden="true"></span>
                                        <span class="d-none d-md-inline"><?php echo Text::_('COM_BANNERS_COUNT_UNPUBLISHED_ITEMS'); ?></span>
                                    </th>
                                <?php endif; ?>
                                <?php if (isset($this->items[0]) && property_exists($this->items[0], 'count_archived')) : ?>
                                    <th scope="col" class="w-10 text-center d-none d-md-table-cell">
                                        <span class="icon-folder icon-fw" aria-hidden="true"></span>
```

**Comment by dennise24** [comment]
I have tested this item :white_check_mark: successfully on 462b155983dcdf2fd4bed720194d12890c448c6d<br /><br />I have tested this item ✅ successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by hannazn** [comment]
I have tested this item :white_check_mark: successfully on 462b155983dcdf2fd4bed720194d12890c448c6d<br /><br />I have tested this item ✅ successfully<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by dorisdreher** [comment]
I have tested this item :red_circle: unsuccessfully on 462b155983dcdf2fd4bed720194d12890c448c6d<br /><br />Error in Articles Categories after installation
An error has occurred.
    500 Layout default_batch_footer not found. 

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by adj9** [comment]
I have not tested this item.<br /><br />In Safari I don't see the difference<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by robertolongo** [comment]
I have tested this item :red_circle: unsuccessfully on 462b155983dcdf2fd4bed720194d12890c448c6d<br /><br />I tested the patch with 5.1.0-alpha1 and ‎5.1.0-alpha4
and I encountered the following error:

An error has occurred.
500 Layout default_batch_footer not found.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by richard67** [comment]
To all who get the error "500 Layout default_batch_footer not found.": How have you applied the pull request and on which Joomla version? That layout has been removed in 5.1 meanwhile.

I will trigger a new build of the testing packages for download right now. Maybe they were not up to date with that change.

You can test again when the downloads are available again.

**Comment by faustonenci** [comment]
I have tested this item :white_check_mark: successfully on 462b155983dcdf2fd4bed720194d12890c448c6d<br /><br />good work<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by peterpeter** [comment]
@richard67 
Joomla 5.1.0-alpha4 with pre-installed com_patchtester from server list on
pizza-bugs-fun.com<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by robertolongo** [comment]
I have tested this item :white_check_mark: successfully on 77281dfad593dabf51ae0402c66c0ec6f02a9671<br /><br />@peterpeter @dorisdreher 
After the drone rebuilder, it works!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by peterpeter** [comment]
I have tested this item :white_check_mark: successfully on 77281dfad593dabf51ae0402c66c0ec6f02a9671<br /><br />Works now :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38787">issues.joomla.org/tracker/joomla-cms/38787</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you!

**Comment by heelc29** [comment]
@chmst @LadySolveig 
The tooltip in the category view only applies to articles and is confusing for all other category views.
image

**Comment by brianteeman** [comment]
Confirmed that the changes in this PR are not correct

**Comment by richard67** [comment]
Then someone should open a new issue for it.

**Comment by chmst** [comment]
Thanks for merging,

@heelc29 you are correct but this is a new issue or PR. 

Will open a new PR for banner clients

**Comment by brianteeman** [comment]
its not a new issue - it is caused by the changes in this pr

**Comment by chmst** [comment]
Sorry, was not quite awake this morning - @brianteeman and @heelc29 were correct 

Please test #42944

**Pull request merged**